cedar-policy/src/api.rs

```diff
@@ -2942,6 +2942,38 @@ impl Policy {
             .collect()
     }
 
+    /// Return a new policy where all occurences of key `EntityUid`s are replaced by value `EntityUid`
+    /// (as a single, non-sequential substitution).
+    pub fn sub_entity_literals(&self, mapping: BTreeMap<EntityUid, EntityUid>) -> Self {
+        // PANIC SAFETY: This can't fail for a policy that was already constructed
+        #[allow(clippy::expect_used)]
+        let cloned_est = self
+            .lossless
+            .est()
+            .expect("Internal error, failed to construct est.")
+            .clone();
+
+        let mapping = mapping.into_iter().map(|(k, v)| (k.0, v.0)).collect();
+
+        // PANIC SAFETY: This can't fail for a policy that was already constructed
+        #[allow(clippy::expect_used)]
+        let est = cloned_est
+            .sub_entity_literals(&mapping)
+            .expect("Internal error, failed to sub entity literals.");
```

Comment:
help me understand why this `expect` is safe -- the panic-safety comment notes that the est must be valid; but what if the mapping isn't valid or something?

Reply:
I think this particular part will be panic-free because we're substituting `EntityUID`s for `EntityUID`s. I'm less confident about the construction of the AST. 

My original thought was that this method can't ensure a policy that previously passed validation still passes after substitution, so the caller will be responsible for checking that policies are valid. I'm not sure that's the best idea. There are some errors that we could catch e.g., mapping `Action::"view"` -> `User::"Alice"`. Maybe we should try to catch as many as possible and then the caller still needs to re-validate against their schema?

Another option would be to require that the mapping not change entity types.

Reply:
If we wanted the mapping to not change entity types, it could be an Eid->Eid mapping instead of EntityUID->EntityUID.  If this fits the need, this might be a better solution.

Reply:
Maybe that's the right call for now? It would work well for the internal use case, but I'm hesitant because I think using the same type for human-readable and `UUID`s is an anti-pattern. It requires adding constraints like disallowing some prefix from the human-readable `Eid` and requiring it for the `UUID`s.